apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java

```diff
@@ -1,81 +1,91 @@
+import org.sagebionetworks.openchallenges.organization.service.model.entity.OrganizationEntity;
+import org.sagebionetworks.openchallenges.organization.service.model.mapper.OrganizationMapper;
+import org.sagebionetworks.openchallenges.organization.service.model.repository.OrganizationRepository;
+
+@ExtendWith(MockitoExtension.class)
+class OrganizationServiceTest {
+
+  @Mock private OrganizationRepository organizationRepository;
+
+  @Mock private OrganizationMapper organizationMapper;
+
+  @BeforeEach
+  void setup() {
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
+  }
+
+  @Test
+  void getOrganization_ShouldReturnOrganizationDto_WhenValidIdentifierPassed() {
+    // Create valid identifier
+    String identifier = "123";
+    Long orgId = 123L;
+    String orgLogin = String.valueOf(identifier);
+
+    OrganizationEntity orgEntity = new OrganizationEntity();
+    orgEntity.setId(orgId);
+
+    OrganizationDto expectedDto = new OrganizationDto();
+    expectedDto.setId(orgId);
+
+    /* Stubbing configuration to simulate a specific behavior of the organizationRepository mock object
+    to control its response and verify the behavior of the tested code that interacts with it. */
+    when(organizationRepository.findByIdOrLogin(orgId, orgLogin))
+        .thenReturn(Optional.of(orgEntity));
+
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
+
+    // get the organization using the identifier
+    OrganizationDto response = organizationService.getOrganization(identifier);
+
+    /* Test that the organization is in the repo, is not null,
+    that the id can be pulled, findByIdOrLogin method is called,
+    and no interactions with the organizationMapper
+    (verifies interactions w/Mock object) */
+
+    verify(organizationRepository).findByIdOrLogin(orgId, orgLogin);
+    assertNotNull(response);
+    assertEquals(expectedDto.getId(), response.getId());
+    verifyNoInteractions(organizationMapper);
+  }
+
+  @Test
+  void getOrganization_ShouldThrowOrganizationNotFoundException_WhenInvalidIdentifierPassed() {
+    // Create invalid identifier
+    String invalidIdentifier = "abc";
+
+    /* Stubbing configuration to simulate a specific behavior of the organizationRepository mock object
+    to control its response and verify the behavior of the tested code that interacts with it. */
+    when(organizationRepository.findByIdOrLogin(any(), any())).thenReturn(Optional.empty());
+
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
+
+    // Test that calling getOrganization throws an error
+    assertThrows(
+        OrganizationNotFoundException.class,
+        () -> {
+          organizationService.getOrganization(invalidIdentifier);
+        });
+    /* Test thatfindByIdOrLogin method is called,
+    and no interactions with the organizationMapper
+    (verifies interactions w/Mock object)*/
```

Comment:
Same comment as above

Reply:
Updated the character count in this line of code.